composer.json

```diff
@@ -20,7 +20,8 @@
         "elasticsearch/elasticsearch": "^7.1.1",
         "nyholm/dsn": "^2.0.0",
         "psr/log": "^1.0",
-        "symfony/deprecation-contracts": "^2.2"
+        "symfony/deprecation-contracts": "^2.2",
+        "symfony/polyfill-php73": "^1.19"
```

Comment:
It seems a bit overkill to have an additional dependency for a single exception change? What is the benefit of having `JsonException`?

Reply:
For preparing smooth upgrade to native `json_encode` and `json_decode` functions with `JSON_THROW_ON_ERROR` option (introduced in PHP 7.3) and fully get rid of `Elastica\JSON` util class and `JSONParseException` exception class.

Reply:
This extra polyfill package can be ignored on project side by using `replace` composer feature.

For example that's what I use in many projects when they are running on modern PHP versions:
```
    "replace": {
        "paragonie/random_compat": "*",
        "ralouphie/getallheaders": "*",
        "symfony/polyfill-apcu": "*",
        "symfony/polyfill-ctype": "*",
        "symfony/polyfill-iconv": "*",
        "symfony/polyfill-intl-grapheme": "*",
        "symfony/polyfill-intl-icu": "*",
        "symfony/polyfill-intl-idn": "*",
        "symfony/polyfill-intl-normalizer": "*",
        "symfony/polyfill-mbstring": "*",
        "symfony/polyfill-php73": "*",
        "symfony/polyfill-uuid": "*"
    },
```